### PR TITLE
Add a form example

### PR DESCRIPTION
**Pull request by lucmirego**
The Form example demonstrate:
* Usage of filter method when using PinLayout's relative methods (above, below, left, right)
* Adjusting a containe's height to match all its children.
* Animation of the appearance/disappearance of UIViews.

### CODE REVIEW

File: Example/PinLayoutSample/UI/Tests/Form/FormView.swift [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+class FormView: BaseFormView {
+    
+    fileprivate let formContainerView = UIView()
+    fileprivate let formTitleLabel = UILabel()
+    fileprivate let nameField = UITextField()
+    
+    fileprivate let ageSwitch = UISwitch()
+    fileprivate let ageField = UITextField()
+    
+    fileprivate let adressField = UITextField()
```
Comment (flambert) [inline Example/PinLayoutSample/UI/Tests/Form/FormView.swift]: addressField (typo)

File: README.md [hunk 2]
```diff
@@ -529,6 +529,34 @@ This is an equivalent solutions using other methods:
 
 <br/>
 
+
+### Positioning using only visible relative UIViews 
+
+All PinLayout's relative methods can accept an array of UIViews (ex: `below(of: [UIView])`). Usign these methods its possible to filter the list of relative UIViews before the list is used by PinLayout.
```
Comment (flambert) [inline README.md]: it's possible (typo)

File: README.md [hunk 3]
```diff
@@ -529,6 +529,34 @@ This is an equivalent solutions using other methods:
...
+
+PinLayout has a filter method called `visible` that can be used to layout a view related to only visible views. This can be really usefull when some views may be hidden for some particular reason.
```
Comment (flambert) [inline README.md]: useful (typo)

File: README.md [hunk 4]
```diff
@@ -529,6 +529,34 @@ This is an equivalent solutions using other methods:
...
+
+```swift
+   formTitleLabel.pin.topCenter().margin(margin)
+   nameField.pin.below(of: formTitleLabel).left().right().height(40).margin(margin)
+        
+   ageSwitch.pin.below(of: nameField).left().right().height(40).margin(margin)
+   ageField.pin.below(of: ageSwitch).left().right().height(40).margin(margin)
+       
+   // Layout the Address UITextField below the last visible view, either ageSwitch or ageField.
+   adressField.pin.below(of: visibles([ageSwitch, ageField])).left().right().height(40).margin(margin)
```
Comment (flambert) [inline README.md]: Meme typo

File: README.md [hunk 5]
```diff
@@ -884,33 +912,36 @@ There is a tiny Example app that expose some usage example on PinLayout, includi
...
-* ...
-
-
 ## FAQ <a name="faq"></a>
 
 *  **Q: When the device rotation change, the layout is not updated.**  
    **R:** PinLayout doesn't use auto layout constraints, it is a framework that manually layout views. For that reason you need to update the layout inside either `UIView.layoutSubviews()` or `UIViewController.viewDidLayoutSubviews()` to handle container size's changes, including device rotation. You'll also need to handle UITraitCollection changes for app's that support multitask.
    
+*  **Q: How can we adjust a UIView container to match all its child?**  
+   **R:** The proposed solution is used by the **Form** example for its rounded corner background. Suppose you want to adjust a container height to match all its childs (subviews).
```
Comment (flambert) [inline README.md]: C'est "its child" dans la question et "its childs" dans la réponse. (je crois que ce doit être "its child" partout)

File: README.md [hunk 6]
```diff
@@ -884,33 +912,36 @@ There is a tiny Example app that expose some usage example on PinLayout, includi
...
 ## FAQ <a name="faq"></a>
 
 *  **Q: When the device rotation change, the layout is not updated.**  
    **R:** PinLayout doesn't use auto layout constraints, it is a framework that manually layout views. For that reason you need to update the layout inside either `UIView.layoutSubviews()` or `UIViewController.viewDidLayoutSubviews()` to handle container size's changes, including device rotation. You'll also need to handle UITraitCollection changes for app's that support multitask.
    
+*  **Q: How can we adjust a UIView container to match all its child?**  
+   **R:** The proposed solution is used by the **Form** example for its rounded corner background. Suppose you want to adjust a container height to match all its childs (subviews). 
+   1. First set the container width and its position:  
+`containerView.pin.topCenter().width(100%).marginTop(10)`
+   2. Layout all its childs.
```
Comment (flambert) [inline README.md]: "its children" ?

File: README.md [hunk 7]
```diff
@@ -920,13 +951,22 @@ This app is available in the `Example` folder. Note that you must do a `pod inst
...
+* minWidth/maxWidth, minHeight/maxHeight
+* ...
+
+
+### Contributing, comments, ideas, suggestions, issues, .... <a name="comments"></a>
 For any **comments**, **ideas**, **suggestions**, **issues**, simply open an [issue](https://github.com/mirego/PinLayout/issues).
 
 If you find PinLayout interresting, thanks to **Star** it. You'll be able to retrieve it easily later.
 
+If you'd like to contribute, your welcome!
```
Comment (flambert) [inline README.md]: "you're" ou "you are"

File: README.md (hunk 6 above)
Comment (lucmirego) [inline README.md]: In fact, the plural of child, its children 👍

File: README.md (hunk 7 above)
Comment (lucmirego) [inline README.md]: Thanks Francois 👍

**Pull request merged**